kernel/sys_clock.c

```diff
@@ -42,7 +42,7 @@ volatile u64_t _sys_clock_tick_count;
  * system clock to track passage of time without interruption.
  * To save power, this should be turned on only when required.
  */
-int _sys_clock_always_on;
+int _sys_clock_always_on = 1;
```

Comment:
where/when is _enable_sys_clock() routine being called by default? (or maybe you are tackling this issue later)

Reply:
It's happening later.  This flag is becoming a hint managed by the kernel.  The driver is responsible for making the decision about what to do with the hardware.  You can take a look at the new systick driver in my timerhacks branch (https://github.com/andyross/zephyr/commit/9296d740ec70e020839c5a21d78955624efac265) to see how it's expected to look.

Basically the kernel decides whether or not to tell the driver to wait for INT_MAX (i.e. give me a tick, but as far in the future as you can) or K_FOREVER (shut it all down, I don't care).